cores/esp8266/HardwareSerial.cpp

```diff
@@ -39,6 +39,10 @@ void HardwareSerial::begin(unsigned long baud, SerialConfig config, SerialMode m
 {
     end();
     _uart = uart_init(_uart_nr, baud, (int) config, (int) mode, tx_pin, _rx_size);
+#if defined(DEBUG_ESP_PORT) && !defined(NDEBUG)
+    println();
+    println(ESP.getFullVersion());

```

Comment:
Unrelated to printing the version, but since you're adding this code block here. Let's call `setDebugOutput(true);` or the equivalent `uart_` function. At the moment debug output only gets enabled before `loop` runs, so you don't get debug output while `setup` is running. Then these three lines can be removed: https://github.com/esp8266/Arduino/blob/7999f5c29c653e6e0b8c77ea42334df8df256a23/cores/esp8266/core_esp8266_main.cpp#L118-L120

Reply:
To understand well, setDebugOutput(true) (= sdk debug messages) is then not automatically enabled anymore when Serial debug is enabled, we would not have espressif debug messages anymore, but only this "full version" displayed and selected arduino-core debug messages ?
For anywone willing to get sdk debug messages, then setDebugOutput(true) has to be called.

That's sound nice to me.

Reply:
I'm suggesting that we move the call to `setDebugOutput(true);` a bit earlier, if DEBUG_ESP_PORT is set. 
Currently it happens only before `loop`, so some debug messages during setup may be lost.
If we do this in HardwareSerial::begin, then we will see all debug messages. So the feature merely becomes more consistent.

(We also need `if (&DEBUG_ESP_PORT == this) {` check before enabling debug output, in case DEBUG_ESP_PORT is the HardwareSerial which we are initializing).

Reply:
Ok to summarize, we **should** automatically enable sdk messages along with core messages, but before setup not after.

OK about Serial/Serial1 distinction.